# Data Science Project - Step 3: Statistical Analysis

**Due: Sunday, April 26 | Points: 25**

In this step, you will:
1. Calculate descriptive statistics
2. Analyze relationships in your data
3. Compare groups (if applicable)
4. Answer your research question with numbers

This builds on Steps 1-2 — use your cleaned data!

In [1]:
import pandas as pd

---
## Setup: Load Your Cleaned Data
---

In [2]:
# Load the cleaned dataset from Step 2
df = pd.read_csv('outputs/cleaned_data.csv')

print(f"Loaded {len(df)} rows")

Loaded 1168 rows


---
## Part 1: Descriptive Statistics (7 points)
---

### Statistics Refresher

| Statistic | What It Tells You | When to Use |
|-----------|------------------|-------------|
| **Mean** (average) | Center of the data | Symmetric data without extreme outliers |
| **Median** (middle value) | Center, resistant to outliers | Data with outliers or skewed distributions |
| **Standard Deviation** | How spread out the data is | Understanding variability |

**When mean ≠ median:**
- Mean > Median: Data is skewed right (some very high values)
- Mean < Median: Data is skewed left (some very low values)
- Mean ≈ Median: Data is roughly symmetric

**Interpreting Standard Deviation:** Compare it to the mean!
- Mean = 100, Std Dev = 5 → Very consistent (5% variation)
- Mean = 100, Std Dev = 50 → Highly variable (50% variation)

In [3]:
# 1a. Summary statistics for numeric columns
df.describe()

,year,wins,losses,winning_percentage,run_differential,runs_scored,runs_allowed
count,1168.000000,1168.000000,1168.000000,1168.000000,1168.000000,1168.000000,1168.000000
mean,1964.623288,79.776541,75.433219,0.513654,21.488014,690.085616,668.597603
std,35.556950,14.855623,14.274299,0.085446,121.844584,116.358028,106.803846
min,1901.000000,19.000000,17.000000,0.248000,-349.000000,219.000000,213.000000
25%,1934.000000,71.000000,66.000000,0.457000,-61.250000,623.750000,607.750000
50%,1966.000000,82.000000,75.000000,0.519500,34.000000,691.000000,671.000000
75%,1995.000000,91.000000,85.000000,0.574000,101.250000,761.250000,734.000000
max,2024.000000,116.000000,120.000000,0.763000,411.000000,1067.000000,1199.000000


In [4]:
# 1b. For your KEY numeric variable(s), calculate and interpret:
#
column = 'winning_percentage'  # Replace with your key numeric variable
mean_val = df[column].mean()
median_val = df[column].median()
std_val = df[column].std()
#
print(f"{column} Statistics:")
print(f"  Mean: {mean_val:.3f}")
print(f"  Median: {median_val:.3f}")
print(f"  Std Dev: {std_val:.2f}")
print(f"  Range: {df[column].min()} to {df[column].max()}")



winning_percentage Statistics:
  Mean: 0.514
  Median: 0.520
  Std Dev: 0.09
  Range: 0.248 to 0.763


### Your Interpretation

**Weak vs. Strong Interpretations:**

| Weak (just describes the number) | Strong (explains what it means) |
|---|---|
| "The mean age is 29.7." | "The average passenger age is 29.7, with a median of 28. These are close, so ages are roughly evenly distributed — no extreme outliers pulling the average." |
| "The standard deviation is 14.5." | "The standard deviation of 14.5 means most passengers are between about 15 and 44 years old — a wide range of ages aboard." |
| "The mean fare is 32.2 and the median is 14.5." | "The average fare ($32.20) is more than double the median ($14.50), which tells us a few very expensive tickets are pulling the average up. Most passengers paid closer to $14.50." |

**Your interpretation:**

the average winning percentage is 0.514 with the median of 0.520 which means most teams' seasons are around the average of 0.500 in baseball terms. the standard deviation is low which means the over performing seasons aren't affecting the average.

In [5]:
# 1c. For your KEY categorical variable(s), calculate:
#     - Value counts (frequency of each category)
#     - Percentages
#
column = 'winning_percentage'
print(f"{column} Frequencies:")
print(df[column].value_counts())
print(f"\n{column} Percentages:")
print(df[column].value_counts(normalize=True).round(3) * 100)

winning_percentage Frequencies:
winning_percentage
0.543    29
0.519    28
0.531    25
0.556    24
0.549    23
         ..
0.433     1
0.400     1
0.317     1
0.660     1
0.685     1
Name: count, Length: 271, dtype: int64

winning_percentage Percentages:
winning_percentage
0.543    2.5
0.519    2.4
0.531    2.1
0.556    2.1
0.549    2.0
        ... 
0.433    0.1
0.400    0.1
0.317    0.1
0.660    0.1
0.685    0.1
Name: proportion, Length: 271, dtype: float64


---
## Part 2: Group Comparisons (8 points)
---

### Comparing Groups — Are Differences Meaningful?

When you compare averages between groups, ask:

1. **How big is the difference?** Compare to the overall range of values.
2. **How much overlap is there?** If std dev is larger than the difference, groups overlap a lot.
3. **How many data points?** More data = more reliable. Be cautious with groups under 30 observations.

**Example:**
> Group A: Mean = 75, Std Dev = 5, N = 500  
> Group B: Mean = 85, Std Dev = 6, N = 450  
> The 10-point difference is LARGE relative to the ~5 point std dev. Group B consistently scores higher. This is a meaningful difference.

In [6]:
# 2a. Group by a categorical variable and calculate statistics
#
# grouped = df.groupby('franchise')['winning_percentage'].agg(['mean', 'median', 'std', 'count'])
grouped = df.groupby('team_name')[['winning_percentage', 'run_differential']].agg(['mean', 'median', 'std', 'count'])
print("Statistics by Group:")
print(grouped)
#
# # Calculate the difference between highest and lowest group
print(f"\nwinning_percentage: Difference between highest and lowest: {grouped[('winning_percentage', 'mean')].max() - grouped[('winning_percentage', 'mean')].min():.2f}")
print(f"Average std dev within groups: {grouped[('winning_percentage', 'std')].mean():.2f}")

print(f"\nrun_differential: Difference between highest and lowest: {grouped[('run_differential', 'mean')].max() - grouped[('run_differential', 'mean')].min():.2f}")
print(f"Average std dev within groups: {grouped[('run_differential', 'std')].mean():.2f}")

Statistics by Group:
                      winning_percentage                          \
                                    mean  median       std count   
team_name                                                          
Atlanta Braves                  0.521678  0.5430  0.080675    59   
Boston Americans                0.541667  0.5715  0.120209     6   
Boston Beaneaters               0.411333  0.3895  0.088753     6   
Boston Bees                     0.466600  0.4610  0.046003     5   
Boston Braves                   0.444971  0.4410  0.080367    35   
Boston Doves                    0.360250  0.3690  0.051565     4   
Boston Red Sox                  0.515436  0.5250  0.078662   117   
Boston Rustlers                 0.291000  0.2910       NaN     1   
Brooklyn Dodgers                0.544143  0.5570  0.090457    28   
Brooklyn Robins                 0.506647  0.4940  0.056797    17   
Brooklyn Superbas               0.431727  0.4340  0.085217    11   
Chicago Cubs               

### Your Group Comparison Findings

Use this format:

> "[Group with highest] has the highest average [measure] at [value], while [Group with lowest] has the lowest at [value]. The difference of [X] is [large/small] compared to the typical variation within groups (std dev of [Y]), suggesting the difference is [meaningful/may be due to random variation]."

**Your findings:**

The New York Yankees have the highest average winning percentage at 0.570, while the Philadelphia Phillies have the lowest at 0.467. The difference of 0.10 is large compared to the typical variation within groups (std dev of ~0.08), suggesting the difference is meaningful.

For run differential, the Yankees lead at a mean of 107.94, while the Philadelphia Phillies sit lowest at -50.80. The difference of ~158 runs is small relative to the very high within-group std devs (~108-130), suggesting run differential varies widely within each franchise.


---
## Part 3: Statistical Testing (5 points)
---

Choose and run **one statistical test** to formally evaluate your research question. Use the flowchart in [Choosing_the_Right_Statistical_Test.html](Choosing_the_Right_Statistical_Test.html) to pick the right test for your data, then copy the code from the matching recipe card.

### Which test do I use?

| Your situation | Test | Guide card |
|----------------|------|-----------|
| Two numeric variables — do they move together? | Correlation (Pearson) | Card 1 |
| Does one numeric variable predict another? | Linear Regression | Card 2 |
| Compare a numeric measure across 2 groups | Independent T-Test | Card 3 |
| Compare your average to a known value | One-Sample T-Test | Card 4 |
| Compare a numeric measure across 3+ groups | ANOVA | Card 5 |
| Are two categorical variables related? | Chi-Square | Card 6 |

**After running your test:** report the p-value and interpret it. If applicable, calculate effect size (Card 7).

### Correlation Strength Reference

| |r| Value | Strength |
|-----------|----------|
| 0.00 - 0.19 | Very weak / no relationship |
| 0.20 - 0.39 | Weak |
| 0.40 - 0.59 | Moderate |
| 0.60 - 0.79 | Strong |
| 0.80 - 1.00 | Very strong |

### CRITICAL: Correlation ≠ Causation!

- Always ask: Could something ELSE explain both variables?
- Write "associated with" or "correlated," NEVER "causes"

In [7]:
# 3a. Choose and run a statistical test
#
# Use the flowchart in Choosing_the_Right_Statistical_Test.html to pick the right test.
# Then copy the code from the matching recipe card and adapt it for your data.

# YOUR TEST HERE

from scipy import stats

r, p_value = stats.pearsonr(df['year'], df['winning_percentage'])
print(f"Correlation (r): {r:.3f}")
print(f"P-value: {p_value:.4f}")


Correlation (r): 0.065
P-value: 0.0270


In [8]:
# 3b. Cross-tabulation for categorical variables (if applicable)
#
# crosstab = pd.crosstab(df['category1'], df['category2'])
# print("Cross-tabulation:")
# print(crosstab)
#
# # Also show percentages (easier to interpret)
# print("\nAs percentages:")
# print(pd.crosstab(df['category1'], df['category2'], normalize='index').round(3) * 100)

### Your Statistical Test Findings

**Which test did you run and why?**

I used a Pearson correlation because I was testing whether a numeric variable (winning percentage) changed over time (year), which involves two numeric variables.

**Interpret your result using the template from the guide:**

The correlation produced an r of 0.065 and a p-value of 0.0270, which is below the 0.05 threshold. This means there is a statistically significant but very weak positive relationship between year and winning percentage.

**Your findings:**

There is a statistically significant but practically negligible correlation between year and winning percentage (r = 0.065, p = 0.027). While the p-value indicates the relationship is unlikely to be due to chance, the r value is so close to zero that year explains almost none of the variation in winning percentage. This confirms that MLB teams have not meaningfully improved over the 123-year period studied.

---
## Part 4: Answer Your Research Question (5 points)
---

### Your Research Question (from Step 1)

How has the average winning percentage of MLB teams changed over time?

State your answer with specific numbers from your analysis:

The average winning percentage of MLB teams has not changed significantly over time. It started at 0.526 in 1901 and was 0.536 in 2024, a difference of only 0.01. While there is year-to-year fluctuation (ranging from a low of 0.471 in 1925 to a high of 0.557 in 2004), there is no consistent upward or downward trend across the 123-year period.

### Was Your Hypothesis Correct?

My hypothesis was that teams would show improved winning percentages over time as the sport became more professionalized and technology improved. This was not supported. Winning percentage has remained stable across franchises, likely because wins and losses are zero-sum across the league, meaning one team's improvement comes at another's expense.

---
## Key Statistics for Your Poster
---

Summarize your key findings here — you'll use these in Step 4 and Step 5.

**Overall Dataset:**
- Total records: 1,169
- Date range: 1901 to 2024

**Main Finding 1:**
- Statistic: The New York Yankees had the highest average winning percentage (0.570) and the Philadelphia Phillies had the lowest (0.467), a difference of 0.10
- What it means: There are meaningful differences in performance across franchises over time

**Main Finding 2:**
- Statistic:  correlation results = 0.065, p = 0.027
- What it means: There is a statistically significant but practically insignificant relationship between year and winning percentage

**Main Finding 3:**
- Statistic: Overall average winning percentage went from 0.526 in 1901 to 0.536 in 2024
- What it means: Winning percentage has remained stable over 123 years with no meaningful upward or downward trend

**Answer to Research Question:** The average winning percentage of MLB teams has not significantly changed over time, remaining close to 0.500 across all franchises from 1901 to 2024.

---
## Submission Checklist
---

Before submitting, verify:

- [ X ] My code runs without errors (Kernel > Restart & Run All)
- [ X ] I calculated descriptive statistics for key variables
- [ X ] I compared groups (if applicable to my question)
- [ X ] I analyzed relationships between variables
- [ X ] I answered my research question with specific numbers
- [ X ] I filled in all the summary sections

**Submit:** This notebook